qcodes/tests/dataset/test_dataset_basic.py

```diff
@@ -680,6 +680,9 @@ def test_parent_dataset_links_invalid_input():
 
     ds = DataSet()
 
+    for link in links:
+        assert link.head != ds.guid
```

Comment:
an idea - could this be biting us https://github.com/QCoDeS/Qcodes/blob/55a8d16bf479e2210833163d664b9ce6cd6cd17a/qcodes/dataset/guids.py#L40 ?

Reply:
Yes that is indeed what I suspect :) Just want to add this to confirm that the issue is indeed identical guids